backend/schema/room.py

```diff
@@ -78,35 +78,69 @@ def validate_dm(cls, values):
             HTTPException [400]: if DM has topic
             HTTPException [400]: if DM has a description
         """
-        plugin_name = values.get("plugin_name")
-        room_members = values.get("room_members", {})
-        topic = values.get("topic")
-        description = values.get("description")
-        created_by = values.get("created_by")
-
-        if plugin_name == Plugin.DM and len(list(room_members.keys())) > 9:
-            raise HTTPException(
-                status_code=400, detail="Group DM cannot have more than 9 members"
-            )
-
-        if plugin_name == Plugin.DM and topic is not None:
-            raise HTTPException(status_code=400, detail="DM should not have a topic")
-
-        if plugin_name == Plugin.DM and description is not None:
-            raise HTTPException(
-                status_code=400, detail="DM should not have a description"
-            )
-
-        if plugin_name == Plugin.DM.value and created_by is not None:
-            raise HTTPException(
-                status_code=400, detail="DM should not have a created by"
-            )
+        if values["room_type"] != RoomType.CHANNEL:
+            room_type = values.get("room_type")
+            room_members = values.get("room_members", {})
+            topic = values.get("topic")
+            description = values.get("description")
+            org_id = values.get("org_id")
+
+            with concurrent.futures.ThreadPoolExecutor(1) as executor:
+                future = executor.submit(
+                    asyncio.run, get_org_rooms(org_id=org_id, room_type=room_type)
+                )
+                rooms = future.result()
+
```

Comment:
Please, can you explain this part?
We would also need comments to make it easier to understand.

Reply:
Fixed.
the validator function cannot be async.
the get_org_rooms is an async function
In order to run an async function in a non async function, the async function has to be ran in a synchronous thread of the the synchronous function.
This handles that

Reply:
Understood. Great.